testsuite/testsuite-webservices/src/test/java/org/wildfly/swarm/webservices/JAXWSClientTest.java

```diff
@@ -0,0 +1,189 @@
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.nio.CharBuffer;
+import java.util.List;
+
+import javax.xml.namespace.QName;
+import javax.xml.ws.BindingProvider;
+import javax.xml.ws.Service;
+import javax.xml.ws.handler.Handler;
+
+import org.jboss.arquillian.container.test.api.Deployment;
+import org.jboss.arquillian.container.test.api.RunAsClient;
+import org.jboss.arquillian.junit.Arquillian;
+import org.jboss.shrinkwrap.api.Archive;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.wildfly.swarm.ContainerFactory;
+import org.wildfly.swarm.config.naming.Binding;
+import org.wildfly.swarm.container.Container;
+import org.wildfly.swarm.naming.NamingFraction;
+import org.wildfly.swarm.undertow.WARArchive;
+import org.wildfly.swarm.webservices.support.SimpleWSServer;
+import org.wildfly.swarm.webservices.web.WebserviceClientServlet;
+import org.wildfly.swarm.webservices.web.WebserviceClientWithHandlerServlet;
+import org.wildfly.swarm.webservices.ws.SampleSoapHandler;
+import org.wildfly.swarm.webservices.ws.SimpleWebserviceEndpointIface;
+import org.wildfly.swarm.webservices.ws.SimpleWebserviceEndpointImpl;
+
+/**
+ * @author sfcoy
+ */
+@RunWith(Arquillian.class)
+public class JAXWSClientTest  implements ContainerFactory {
+
+    private static final String MESSAGE_TO_ECHO = "Lorem ipsum dolor sit amet";
+    private static final QName SERVICE_Q_NAME = new QName(NAMESPACE, SERVICE_NAME);
+
+    private static SimpleWSServer wsServer;
+
+    public JAXWSClientTest() {
+        System.out.println(" --> Created JAXWSClientTest");
+        System.out.flush();
+    }
+
+    @Deployment(testable = false)
+    public static Archive createDeployment() {
+        System.out.println("Creating deployment");
+        final WARArchive warArchive = ShrinkWrap.create(WARArchive.class, "myapp.war")
+                .addClass(WebserviceClientServlet.class)
+                .addClass(WebserviceClientWithHandlerServlet.class)
+                .addClass(SampleSoapHandler.class)
+                .addClass(SimpleWSServer.class)
+                .addClass(SimpleWebserviceEndpointImpl.class)
+                .addClass(SimpleWebserviceEndpointIface.class);
+        warArchive.addModule("javax.xml.ws.api");
+        return warArchive;
+    }
+
+    public static void startUp() throws IOException {
+        wsServer.buildAndStartWebService();
+    }
+
+    public Container newContainer(String... args) throws Exception {
+        wsServer = new SimpleWSServer();
+        wsServer.buildAndStartWebService();
+        System.out.println("Building container");
+        return new Container(false)
+                .fraction(WebServicesFraction.createDefaultFraction())
+                .fraction(buildNamingFraction());
+    }
+
```

Comment:
Ignore the two tests below, they require the presence of an @Before, @After methods that set up and take down the web service. They just prove that the basic machinery works out of container